lxd/instance/drivers/driver_lxc.go

```diff
@@ -8507,6 +8507,9 @@ func (d *lxc) getFSStats() (*metrics.MetricSet, error) {
 
 	for scanner.Scan() {
 		fields := strings.Fields(scanner.Text())
+		if len(fields) < 3 {
+			continue
+		}
```

Comment:
Can we avoid using a magic number here? Or, add a comment explaining why we expect `len(fields)` to be at least 3.

Reply:
The line right after accesses `fields[2]` so the surrounding context dispels some of the magic, IMHO :)

Maybe you'd find this more readable?

```
	for scanner.Scan() {
		fields := strings.Fields(scanner.Text())
		if len(fields) >= 2 {
			mountMap[fields[0]] = mountInfo{Mountpoint: fields[1], FSType: fields[2]}
		}
	}
```

I'm ambivalent personally.

Reply:
Although the surrounding context is useful, the magic number can lead to future bugs if there is ever an unlikely scenario where `fields` changes. Something like this would be nice:

```suggestion
	// Fields in /proc/mounts lines are space-separated, e.g.:
	//   source target fstype options dump pass
	const (
		fieldSource    = 0
		fieldTarget    = 1
		fieldFSType    = 2
		minMountFields = 3 // we only need the first three
	)

	for scanner.Scan() {
		fields := strings.Fields(scanner.Text())
		if len(fields) < minMountFields {
			continue
		}

		mountMap[fields[fieldSource]] = mountInfo{
			Mountpoint: fields[fieldTarget],
			FSType:     fields[fieldFSType],
		}
	}
```

This also acts as documentation for us :)

Reply:
Just a comment describing the fields also works.

Reply:
Hmm, while that's definitely more idiomatic I feel like it goes against the style of the surrounding code and makes it stick out a bit. So unless you feel strongly about it, I'd keep the magic `3`.

Reply:
Yeah for me, 3 isn't a magic number here anyway, its clearly the required length of the slice because we want that field.

A magic number in my view is something that we send/receive into an API that has meaning outside of its numeric value.